plugins/modules/cloud/misc/terraform.py

```diff
@@ -449,12 +474,41 @@ def main():
     if state == 'present' and module.params.get('parallelism') is not None:
         command.append('-parallelism=%d' % module.params.get('parallelism'))
 
+    def process_complex_args(vars):
+        ret_out = []
+        if isinstance(vars, dict):
+            for k, v in vars.items():
+                if isinstance(v, dict):
+                    ret_out.append('{0}={{{1}}}'.format(k, process_complex_args(v)))
+                if isinstance(v, list):
+                    ret_out.append("{0}=[{1}]".format(k, process_complex_args(v)))
+                if isinstance(v, (integer_types, float, str)):
+                    ret_out.append('{0}={1}'.format(k, json.dumps(v)))
+        if isinstance(vars, list):
+            l_out = []
+            for item in vars:
+                if isinstance(item, dict):
+                    l_out.append("{{{0}}}".format(process_complex_args(item)))
+                elif isinstance(item, (str, integer_types, float)):
+                    l_out.append(json.dumps(item))
+                else:
+                    module.fail_json(msg="List can contain, dictionaries, strings, integer_types and float.")
```

Comment:
I never used list of lists in ansible. For example:

```
- this is top level item1
   - this is a second level item 1
   - this is a second level item 2
   - this is a second level item 3
- this is top level item2
- this is top level item3
```

Testing with the following sample:

```
- hosts: localhost
  gather_facts: false
  tasks: 
    - debug:
        var: listoflists[0][0]

  vars:
    listoflists: 
      - this is top level item1
        - this is a second level item 1
        - this is a second level item 2
        - this is a second level item 3
      - this is top level item2
      - this is top level item3
```

I would expect debug to print ``` this is a second level item 1```, but I get the following.

```
TASK [debug] ***********************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "listoflists[0][0]": "t"
}
```
Does the above test make any sense, or even possible? If this is a valid Ansible interpretation, I'm happy to map that in. I can think of scenarios how this would work in Terraform, but I doubt Ansible can make that interpretation. Hence, I skipped that scenario :).

Reply:
This is because you haven't specified a list of lists, but a list of strings, where the first list element is a string that contains multiple `-`s.

You should try
```.yaml
  vars:
    listoflists: 
      - - this is a second level item 1
        - this is a second level item 2
        - this is a second level item 3
      - this is top level item2
      - this is top level item3
```
instead.

Reply:
This is added. List of lists is now supported.